src/pages/VaccinePage.js

```diff
@@ -10,128 +12,66 @@ import {
   ChevronRightIcon,
   PageButton,
   Paper,
-  BatteryIcon,
-  DownloadIcon,
   TemperatureIcon,
-  IconButton,
-  CogIcon,
   Circle,
   FlexRow,
   AlarmClockIcon,
-  WifiIcon,
 } from '../widgets/index';
 import { TextWithIcon } from '../widgets/Typography';
 
-import { buttonStrings } from '../localization';
-import { DARKER_GREY, BLACK, FINALISE_GREEN } from '../globalStyles';
-import { gotoEditSensorPage, gotoFridgeDetailPage, gotoNewSensorPage } from '../navigation/actions';
+import { generalStrings, buttonStrings } from '../localization';
+import { DARKER_GREY, BLACK } from '../globalStyles';
+import { gotoFridgeDetailPage, gotoNewSensorPage } from '../navigation/actions';
 import { AfterInteractions } from '../widgets/AfterInteractions';
-import { BlinkSensorButton } from '../widgets/BlinkSensorButton';
+import { FridgeHeader } from '../widgets/FridgeHeader';
+import { selectSensors } from '../selectors/Entities/sensor';
+import temperature from '../utilities/temperature';
 
-const formatDate = date => moment(date).fromNow();
-const formatTemperature = temperature => `${Math.round(temperature * 10) / 10}°C`;
+const formatDate = date => (date ? moment(date).fromNow() : generalStrings.not_available);
 
-const FridgeDisplay = ({ fridge, toFridgeDetail, toEditSensorPage }) => {
-  const { description, sensors } = fridge;
-  const [sensor] = sensors;
-  const header = (
-    <>
-      <TextWithIcon
-        left
-        size="ms"
-        textStyle={{ textTransform: 'uppercase' }}
-        containerStyle={{ flex: 1 }}
-        Icon={<Circle size={20} backgroundColor={FINALISE_GREEN} />}
-      >
-        {description}
-      </TextWithIcon>
-      <IconButton Icon={<DownloadIcon color={BLACK} />} containerStyle={localStyles.iconButton} />
-      <BlinkSensorButton macAddress={sensor?.macAddress} />
-      <IconButton
-        Icon={<CogIcon color={BLACK} />}
-        onPress={() => toEditSensorPage(sensor)}
-        containerStyle={localStyles.iconButton}
-      />
-    </>
-  );
+const FridgeDisplay = ({ sensor, toFridgeDetail }) => {
+  const { id, macAddress, currentTemperature, mostRecentBreachTime, locationID } = sensor;
 
   return (
     <Paper
-      // macAddress passed to fridgeHeader is quite naff. The settings page for the "fridge" is
-      // actually for a sensor, assuming 1 sensor per fridge.
-      Header={header}
+      Header={<FridgeHeader showTitle showCog sensor={sensor} />}
```

Comment:
Having a look now and it would be possible to pass just the `sensorID` to `FridgeHeader` and `FridgeHeader` itself can pick up all the values it needs since it's already connected to redux, by just using the `ID`. This might be a better solution as passing `sensor` as a prop will cause some unneeded re-renders when the object is mutated rather than passing only the primitives which are de structured ...

Reply:
🤣 


7f10cfd